### PR TITLE
Droth 1152 direction of traffic in roundabouts

### PR DESCRIPTION
**Pull request by joanacoelho**

### CODE REVIEW

File: digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/util/RoundaboutProcessor.scala [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+case class RoundaboutChangeSet(trafficDirectionChanges: Seq[RoadLink] = Seq.empty)
+
+object RoundaboutProcessor {
+
+  private def findRoundaboutRecursive(firstLink: RoadLink, nextLink: RoadLink, roadLinks: Seq[RoadLink], acc: Seq[RoadLink] = Seq.empty, withIncomplete: Boolean = true): (Seq[RoadLink], Seq[RoadLink]) = {
+    val (adjacents, rest) = roadLinks.partition(r => GeometryUtils.areAdjacent(r.geometry, nextLink.geometry))
+    adjacents match{
+      case Seq() if acc.size < 3 && !withIncomplete => {
+        println(s"findRoundaboutRecursive: ")
```
Comment (leirasf) [inline digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/util/RoundaboutProcessor.scala]: We should use the logger. And it the message should be incomplete roundabout found.

File: digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/util/RoundaboutProcessor.scala [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+import fi.liikennevirasto.digiroad2.{GeometryUtils, Point}
+
+case class RoundaboutChangeSet(trafficDirectionChanges: Seq[RoadLink] = Seq.empty)
+
+object RoundaboutProcessor {
+
+  private def findRoundaboutRecursive(firstLink: RoadLink, nextLink: RoadLink, roadLinks: Seq[RoadLink], acc: Seq[RoadLink] = Seq.empty, withIncomplete: Boolean = true): (Seq[RoadLink], Seq[RoadLink]) = {
+    val (adjacents, rest) = roadLinks.partition(r => GeometryUtils.areAdjacent(r.geometry, nextLink.geometry))
+    adjacents match{
+      case Seq() if acc.size < 3 && !withIncomplete => {
```
Comment (leirasf) [inline digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/util/RoundaboutProcessor.scala]: It should be added the possibility to have roundabout with only 1 link.

File: digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/dao/RoadLinkDAO.scala [hunk 3]
```diff
@@ -0,0 +1,297 @@
...
+
+  def insertValues(linkProperty: LinkProperties, vvhRoadlink: VVHRoadlink, username: Option[String], value: Int, mmlId: Option[Long]): Unit = {
+    sqlu"""insert into #$table (id, link_id, #$column, modified_by )
+                   select primary_key_seq.nextval, ${linkProperty.linkId}, ${value}, $username
+                   from dual
+                   where not exists (select * from #$table where link_id = ${linkProperty.linkId})""".execute
+  }
+
+  def insertValues(linkProperty: LinkProperties, username: Option[String], value: Int): Unit = {
+    sqlu"""insert into #$table (id, link_id, #$column, modified_by )
```
Comment (joanacoelho) [inline digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/dao/RoadLinkDAO.scala]: This method can call the above one

File: digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/dao/RoadLinkDAO.scala [hunk 4]
```diff
@@ -0,0 +1,297 @@
...
+  def updateValues(linkProperty: LinkProperties, vvhRoadlink: VVHRoadlink, username: Option[String], value: Int, mml_id: Option[Long]): Unit = {
+      sqlu"""update #$table
+               set #$column = $value,
+                   modified_date = SYSDATE,
+                   modified_by = $username
+               where link_id = ${linkProperty.linkId}""".execute
+  }
+
+  def updateValues(linkProperty: LinkProperties, username: Option[String], value: Int): Unit = {
+    sqlu"""update #$table
```
Comment (joanacoelho) [inline digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/dao/RoadLinkDAO.scala]: This method can call the above one!

**Pull request merged**